# HG changeset patch
# User Rik <rik@octave.org>
# Date 1406386238 25200
#      Sat Jul 26 07:50:38 2014 -0700
# Branch stable
# Node ID a5add7b660acd575fb204161c9993daa9abd3e34
# Parent  87cba451fd5eed2f0105b5be67fe57a708e1aac4
build: Fix compiling GUI on Alpha platforms (bug #42839).

* libgui/Makefile.am: Use GNU Makefile filter-out command to remove
-mieee CFLAG option from CPPFLAGS passed to MOC command.

diff --git a/libgui/Makefile.am b/libgui/Makefile.am
--- a/libgui/Makefile.am
+++ b/libgui/Makefile.am
@@ -17,16 +17,20 @@
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
 MOC_CPPFLAGS =
 
+## Fix for bug #42839 where -mieee CFLAG option is added to CPPFLAGS by gnulib.
+## Eventually gnulib will be appropriately fixed and this hack removed.
+MOC_OCTAVE_CPPFLAGS = $(filter-out -mieee, $(AM_CPPFLAGS) $(CPPFLAGS))
+
 octlib_LTLIBRARIES = liboctgui.la
 
 TRANSLATIONS = \
   languages/be_BY.ts \
   languages/de_DE.ts \
   languages/en_US.ts \
   languages/es_ES.ts \
   languages/fr_FR.ts \
@@ -86,17 +90,17 @@ octlocale_DATA = $(LOCALES)
 
 default-qt-settings: default-qt-settings.in
 	$(do_subst_qt_settings)
 
 define moc-command
 ( echo '#ifdef HAVE_CONFIG_H'; \
   echo '#include <config.h>'; \
   echo '#endif'; \
-  $(MOC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(MOC_CPPFLAGS) $(liboctgui_la_CPPFLAGS) $< ) > $@-t
+  $(MOC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(MOC_OCTAVE_CPPFLAGS) $(MOC_CPPFLAGS) $(liboctgui_la_CPPFLAGS) $< ) > $@-t
 mv $@-t $@
 endef
 
 define rcc-command
 ( echo '#ifdef HAVE_CONFIG_H'; \
   echo '#include <config.h>'; \
   echo '#endif'; \
   $(RCC) $< ) > $@-t
